### ISSUE TITLE
AWS Background Sync causes ingress statuses to flush

### ISSUE THREAD

**Issue by joshrosso** [coreos/alb-ingress-controller#220]
With the ALB background sync introduced in #197, after each sync the ingress status gets sets back to some cluster IP rather than retaining the ALB's hostname. 

Placeholder for now, more details to come.

**Comment by joshrosso** [comment]
After one of the AWS recycles occurs, you'll see the following flip between some cluster ips to the ALB IP.

```
I0921 21:20:39.485810    5334 albingress.go:268] [ALB-INGRESS] [echoserver/echoserver3] [INFO]: Ingress rebuilt from existing ALB in AWS
I0921 21:20:39.485847    5334 albingresses.go:150] [ALB-INGRESS] [ingresses] [INFO]: Assembled 1 ingresses from existing AWS resources
I0921 21:20:41.025864    5334 status.go:337] updating Ingress echoserver/echoserver3 status to [{10.0.51.249 } {10.0.90.49 } {18.221.101.58 } {18.221.105.0 }]
I0921 21:20:41.159360    5334 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"echoserver", Name:"echoserver3", UID:"246a919c-9e0f-11e7-afe6-021afa6d10a4", APIVersion:"extensions", ResourceVersion:"1959859", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress echoserver/echoserver3
I0921 21:21:41.227815    5334 status.go:337] updating Ingress echoserver/echoserver3 status to [{ joshcalico-echoserver-echo-ac22-470386820.us-east-2.elb.amazonaws.com}]
I0921 21:21:41.356722    5334 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"echoserver", Name:"echoserver3", UID:"246a919c-9e0f-11e7-afe6-021afa6d10a4", APIVersion:"extensions", ResourceVersion:"1960051", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress echoserver/echoserver3
```

Not sure why yet.

**Comment by joshrosso** [comment]
Found the issue, since we introduced the reconcile flag (https://github.com/coreos/alb-ingress-controller/pull/197/commits/fd79f6a6c991aca2b262edd0c939824c87d0cd5a) to ensure that failed to reconcile ingresses don't have their status updated (https://github.com/coreos/alb-ingress-controller/issues/191) we now check the reconciled flag before returning the hostname.

After the ALB ingress sync, the reconciled flag isn't set to true, thus nil is returned and the cluster IPs propagate up to the status (until next reconcile kicks off and fixes it). Sending an update now to #197 and cutting `1.0-alpha.2`.
